app/src/main/java/com/framgia/soundclound/data/source/TrackRepository.java

```diff
@@ -1,26 +1,36 @@
 package com.framgia.soundclound.data.source;
 
-import android.support.annotation.NonNull;
-
-import com.framgia.soundclound.data.model.Track;
-
-import java.util.List;
-
 /**
  * Created by Sony on 1/5/2018.
  */
 
-public class TrackRepository implements TrackDataSource {
+public class TrackRepository {
```

Comment:
WARNING: Class TrackRepository should be declared as final.

Reply:
thắc mắc vì  sao lại đặt là final ??